### PR TITLE
Pulse translator

### PR DESCRIPTION
**Pull request by eggerdj**
### Summary
PR with a translator to translate pulse schedules into signals.

### Details and comments

* Added the `InstructionsToSignals` converter
* Unit tests
* Demo NB.

### CODE REVIEW

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            ch = inst.channel.name
+            phi = phases[ch]
+            freq = frequency_shifts[ch]
+
+            if isinstance(inst, Play):
+                samples = []
+                start_idx = len(signals[ch].samples)
+                for idx, sample in enumerate(inst.pulse.get_waveform().samples):
+                    t = self._dt * (idx + start_idx)
+                    samples.append(sample * np.exp(1.0j * freq * t + 1.0j * phi))
```
Comment (DanPuzzuoli) [inline qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py]: Should this be `2 * np.pi * freq`?

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+                signals[ch].add_samples(start_time, samples)
```
Comment (DanPuzzuoli) [inline qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py]: Incredibly minor but could you change the `start_time` variable to `start_sample` to be consistent with the `add_samples` method signature? Or is `start_time` more consistent with what things are called in Pulse?

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import numpy as np
```
Comment (DanPuzzuoli) [inline qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py:17]: Add copyright blurb

File: test/terra/pulse_new/models/test_pulse_to_signals.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
```
Comment (DanPuzzuoli) [inline test/terra/pulse_new/models/test_pulse_to_signals.py]: Add 2020

File: test/terra/pulse_new/models/test_pulse_to_signals.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        converter = InstructionToSignals(dt=1, carriers=None)
+
+        signals = converter.get_signals(sched)
+
+        self.assertEquals(len(signals), 2)
+        self.assertTrue(isinstance(signals[0], PiecewiseConstant))
+        self.assertTrue(isinstance(signals[0], PiecewiseConstant))
+
+        samples = test_gaussian.get_waveform().samples
+        self.assertTrue(np.allclose(signals[1].samples[0:len(samples)], samples))
```
Comment (DanPuzzuoli) [inline test/terra/pulse_new/models/test_pulse_to_signals.py:51]: In relation to the comment about frequencies, can you add a check here that the frequency shift on `DriveChannel(0)` is behaving correctly?

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                phases[ch] += inst.phase
+
+            if isinstance(inst, ShiftFrequency):
+                frequency_shifts[ch] += inst.frequency
+
+            if isinstance(inst, SetPhase):
+                phases[ch] = inst.phase
+
+            if isinstance(inst, SetFrequency):
+                frequency_shifts[ch] = inst.frequency
```
Comment (DanPuzzuoli) [inline qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py]: Should this be `inst.frequency - carrier_freq`?

**Review by DanPuzzuoli — CHANGES_REQUESTED** [review]
Aside from clarifying the frequency handling (and other minor comments), looks good overall.

Additionally, can you delete the `pulse_new/models/pulse_converter.py` in this PR? (I assume it is out of date and replaced by `converters/pulse_to_signals.py`.

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py (hunk 1 above)
Comment (eggerdj) [reply qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py]: Good catch. Thanks.

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py (hunk 2 above)
Comment (eggerdj) [reply qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py]: Good point. Thanks.

File: qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py (hunk 6 above)
Comment (eggerdj) [reply qiskit/providers/aer/pulse_new/converters/pulse_to_signals.py]: Yes. Thanks!

**Review by DanPuzzuoli — APPROVED** [review]
Awesome, looks good, will merge.

**Pull request merged**